src/cmd/services/m3coordinator/ingest/carbon/ingest.go

```diff
@@ -68,12 +68,14 @@ var (
 	errCannotGenerateTagsFromEmptyName = errors.New("cannot generate tags from empty name")
 	errIOptsMustBeSet                  = errors.New("carbon ingester options: instrument options must be st")
 	errWorkerPoolMustBeSet             = errors.New("carbon ingester options: worker pool must be set")
+	errMultipleWorkerPoolsSet          = errors.New("carbon ingester options: only single worker pool can be set")
 )
 
 // Options configures the ingester.
 type Options struct {
 	InstrumentOptions instrument.Options
-	WorkerPool        xsync.PooledWorkerPool
+	StaticWorkerPool  xsync.StaticPooledWorkerPool
```

Comment:
this was only code that allowed switching between static and dynamic. in order to get rid of the `GrowOnDemand` option, I needed to inline both pools. I don't think this is a pattern we should encourage, so explicitly having to toggle seemed like the best approach.